synapse/lib/fifo.py

```diff
@@ -0,0 +1,389 @@
+import os
+import struct
+import logging
+import threading
+import collections
+
+import synapse.glob as s_glob
+import synapse.common as s_common
+import synapse.eventbus as s_eventbus
+
+import synapse.lib.atomic as s_atomic
+import synapse.lib.config as s_config
+import synapse.lib.msgpack as s_msgpack
+import synapse.lib.atomfile as s_atomfile
+
+logger = logging.getLogger(__file__)
+
+class Fifo(s_config.Config):
+
+    def __init__(self, conf, xmit=None):
+
+        s_config.Config.__init__(self)
+        self.setConfOpts(conf)
+        self.reqConfOpts()
```

Comment:
true, but since the xmit function is required, there wasn't really much hope of making asingle Fifo from a dmon anyway :)  The idea here is that a parent object will have the config for this.